Filter by radius

LogicX1 opened the pull request:
This needs changes here and there, and to fine tune numbers or use a more accurate method.
Putting it here so we all can see it.

Adds an input range to choose radius, radius is a state in home page.
Adds a circle around the given radius (google maps circle takes radius in meters).
Changing the radius will redraw the Circle and change the zoom level of the map( it is not accurate right now ).

rel #4  : Stretch goal to filter by distance

LogicX1 committed:
Adds radius input in filtering menu
In Footer => Filtering : added a range slider to choose a radius
This slider will change the state radius at Home component.
Styling the range input
rel #4

LogicX1 committed:
Adds a circle around the chosen radius, and zooms around the radius
This not working 100% , the value that i determine for the zoom is not accurate, needs more testing and finding better numbers or another way to do it

Karyum reviewed (commented):
So let's change a bit in the functionality of this:
1) don't show circle if the user hasn't choose radius yet.
2) limit the radius to a maximum of 5km

Karyum commented on src/components/BusinessesMap/example.jsx:
```diff
@@ -0,0 +1,103 @@
+
+const MapWithASearchBox = compose(
+    withProps({
+        googleMapURL: 'https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places',
```
not a very big deal since the the api_key can be found in the browser, but let's keep the code consistent by keeping the key as an environment variable

LogicX1 replied:
Oh sorry this file will be deleted, it is an example I got from the documentation to help who ever is working on it, but its very different from what im using i will delete it.

Karyum commented on src/components/BusinessesMap/example.jsx:
```diff
@@ -0,0 +1,103 @@
+            this.setState({
+                bounds: null,
+                center: {
+                    lat: 41.9, lng: -87.624
```
why is it hardcoded?

Karyum commented on src/pages/Home/index.jsx:
```diff
@@ -67,6 +68,9 @@ export default function Homepage(props) {
 
+    useEffect(()=>{
+        console.log(radius);
+    },[radius])
```
we can delete this btw

LogicX1 committed:
Fixes few linter errors

LogicX1 committed:
Adressed the changes requested in the PR
Circle will only show after we filter by radius
Circle will clear when we clear the filteration
Radius range changed to 0.5-5KM , with steps of 0.5 KM
Zoom now will fit the radius better (Its a bit hard to fit the circle exactly)
rel #4
